concept/type_/relation_type.rs

```
@@ -455,9 +454,66 @@ impl RelationType {
         Ok(None)
     }
 
-    get_with_specialised! {
-        pub fn get_relates_role_with_specialised() -> Relates = RoleType | get_relates_role;
-        pub fn get_relates_role_name_with_specialised() -> Relates = &str | get_relates_role_name;
+    pub fn get_relates_role_explicit_declared(
+        &self,
+        snapshot: &impl ReadableSnapshot,
+        type_manager: &TypeManager,
+        role_type: RoleType,
+    ) -> Result<Option<Relates>, Box<ConceptReadError>> {
+        if let Some(relates) = self.get_relates_role_declared(snapshot, type_manager, role_type.clone())? {
+            if !relates.is_implicit(snapshot, type_manager)? {
+                return Ok(Some(relates));
+            }
+        }
+        Ok(None)
+    }
+
+    pub fn get_relates_role_explicit(
```

I still have the non-declared explicit versions for other places in definition resolution. E.g., when we define
```
define
parentship relates parent;
fathership relates father as parent;
```

When we say
```
fathership relates parent;
```
^ this should return an error. However, if we don't search for **explicit** relates here, we'd find an implicit relates, its definition status would be "just like the user defined!", and the query would do nothing, saying "OK!". And the user won't even understand that his request is invalid.

Forgot to mention it in person. However, I do believe that the usage of implicit / explicit already makes everything much cleaner and streamlined.

I just don't like the duplications for "_role_" and "_role_name_", would be cool to find a way to generalize it (if needed).